### PR TITLE
[RFS-50] Add support for returnDataSize, returnDataCopy and extCodeSize for precompiled contracts

### PR DESCRIPTION
**Pull request by diega**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 151 issues
* ![MAJOR][MAJOR] 82 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 60 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [VM.java#](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#): This file has 1,501 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [VM.java#L182](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L182): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [VM.java#L209](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L209): Method org.ethereum.vm.VM.doADD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L226](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L226): Method org.ethereum.vm.VM.doMUL() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L242](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L242): Method org.ethereum.vm.VM.doSUB() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L258](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L258): Method org.ethereum.vm.VM.doDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L274](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L274): Method org.ethereum.vm.VM.doSDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L290](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L290): Method org.ethereum.vm.VM.doMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L306](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L306): Method org.ethereum.vm.VM.doSMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L327](https://github.com/rsksmart/rskj/blob/72e6ae891afd3251c6683e56ed49a9aecf27e365/rskj-core/src/main/java/org/ethereum/vm/VM.java#L327): Method org.ethereum.vm.VM.doEXP() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/vm/VM.java [hunk 1]
```diff
@@ -767,20 +768,23 @@ protected void doCODESIZE() {
...
-        {
-            length = program.getCode().length; // during initialization it will return the initialization code size
+        DataWord codeLength;
+        if (op == OpCode.CODESIZE) {
+            codeLength = new DataWord(program.getCode().length); // during initialization it will return the initialization code size
         } else {
             DataWord address = program.stackPop();
-            length = program.getCodeAt(address).length;
+            codeLength = new DataWord(program.getCodeAt(address).length);
+            if (program.getBlockchainConfig().isRfs90()) {
```
Comment (amendelzon) [inline rskj-core/src/main/java/org/ethereum/vm/VM.java]: `isRfs50`?

File: rskj-core/src/main/java/org/ethereum/vm/VM.java [hunk 2]
```diff
@@ -767,20 +768,23 @@ protected void doCODESIZE() {
...
+                PrecompiledContracts.PrecompiledContract precompiledContract = PrecompiledContracts.getContractForAddress(config, address);
+                if (precompiledContract != null) {
+                    codeLength = new DataWord(BigIntegers.asUnsignedByteArray(DataWord.MAX_VALUE));
```
Comment (amendelzon) [inline rskj-core/src/main/java/org/ethereum/vm/VM.java]: I thought we agreed on using a small number, e.g., `1`

File: rskj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 3]
```diff
@@ -1573,6 +1574,10 @@ public void callToPrecompiledAddress(MessageCall msg, PrecompiledContract contra
 
             byte[] out = contract.execute(data);
 
+            if (getBlockchainConfig().isRfs90()) {
```
Comment (amendelzon) [inline rskj-core/src/main/java/org/ethereum/vm/program/Program.java]: `isRfs50`?

File: rskj-core/src/main/java/org/ethereum/vm/program/Program.java [hunk 4]
```diff
@@ -1573,6 +1574,10 @@ public void callToPrecompiledAddress(MessageCall msg, PrecompiledContract contra
...
+                this.returnDataBuffer = out;
```
Comment (amendelzon) [inline rskj-core/src/main/java/org/ethereum/vm/program/Program.java]: Is this all that's needed to implement `returnDataSize` and `returnDataCopy`?

File: rskj-core/src/main/java/org/ethereum/vm/VM.java (hunk 1 above)
Comment (diega) [reply rskj-core/src/main/java/org/ethereum/vm/VM.java]: You are right! thanks for checking this!

File: rskj-core/src/main/java/org/ethereum/vm/VM.java (hunk 2 above)
Comment (diega) [reply rskj-core/src/main/java/org/ethereum/vm/VM.java]: Yes, I'm sorry, I should have put this into the comments section of the PR. While I was implementing this I asked @donequis for his guidance and this matter came up. He told me, and I agreed, that at some point this value could be used for something different than just checking if there is a contract at a given address or not, and a small number here may allow that usage in the context of a precompiled contract while a really big number shouldn't

File: rskj-core/src/main/java/org/ethereum/vm/program/Program.java (hunk 4 above)
Comment (diega) [reply rskj-core/src/main/java/org/ethereum/vm/program/Program.java]: Yes, that's all. I'll add another test for checking `returnDataCopy` explicitly b/c atm I'm just checking `returnDataSize`

**Comment by rskops** [comment]
SonarQube analysis reported 151 issues
* ![MAJOR][MAJOR] 82 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 60 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [VM.java#](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#): This file has 1,501 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [VM.java#L182](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L182): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [VM.java#L209](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L209): Method org.ethereum.vm.VM.doADD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L226](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L226): Method org.ethereum.vm.VM.doMUL() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L242](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L242): Method org.ethereum.vm.VM.doSUB() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L258](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L258): Method org.ethereum.vm.VM.doDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L274](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L274): Method org.ethereum.vm.VM.doSDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L290](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L290): Method org.ethereum.vm.VM.doMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L306](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L306): Method org.ethereum.vm.VM.doSMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L327](https://github.com/rsksmart/rskj/blob/5238b90968c32180f376b924f6f95aecdc1279b5/rskj-core/src/main/java/org/ethereum/vm/VM.java#L327): Method org.ethereum.vm.VM.doEXP() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/vm/VM.java [hunk 5]
```diff
@@ -767,20 +768,23 @@ protected void doCODESIZE() {
...
-            length = program.getCode().length; // during initialization it will return the initialization code size
+        DataWord codeLength;
+        if (op == OpCode.CODESIZE) {
+            codeLength = new DataWord(program.getCode().length); // during initialization it will return the initialization code size
         } else {
             DataWord address = program.stackPop();
-            length = program.getCodeAt(address).length;
+            codeLength = new DataWord(program.getCodeAt(address).length);
+            if (program.getBlockchainConfig().isRfs50()) {
+                PrecompiledContracts.PrecompiledContract precompiledContract = PrecompiledContracts.getContractForAddress(config, address);
```
Comment (donequis) [inline rskj-core/src/main/java/org/ethereum/vm/VM.java]: Can we add a method that just checks whether a contract exists at an address?

**Comment by rskops** [comment]
SonarQube analysis reported 151 issues
* ![MAJOR][MAJOR] 82 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 60 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [VM.java#](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#): This file has 1,501 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [VM.java#L182](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L182): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [VM.java#L209](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L209): Method org.ethereum.vm.VM.doADD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L226](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L226): Method org.ethereum.vm.VM.doMUL() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L242](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L242): Method org.ethereum.vm.VM.doSUB() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L258](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L258): Method org.ethereum.vm.VM.doDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L274](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L274): Method org.ethereum.vm.VM.doSDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L290](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L290): Method org.ethereum.vm.VM.doMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L306](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L306): Method org.ethereum.vm.VM.doSMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L327](https://github.com/rsksmart/rskj/blob/6886ae24686ff3a624884973fca0ceb39d226abb/rskj-core/src/main/java/org/ethereum/vm/VM.java#L327): Method org.ethereum.vm.VM.doEXP() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by diega** [comment]
Yes, @josedahlquist, that would be great. Unfortunately there isn't way to do this in plain java, if we wanted to do that we should invoke the actual command from the system, causing it other problems. If there is enough interest we could evaluate implementing the compiler on top of [antlr](http://www.antlr.org/), for whom there is already a proper grammar in [solidityj](https://github.com/solidityj/solidity-antlr4)

File: rskj-core/src/main/java/org/ethereum/vm/VM.java (hunk 5 above)
Comment (diega) [reply rskj-core/src/main/java/org/ethereum/vm/VM.java]: Would you mind if I add this request as a future pull request/improvement? I would like to move this forward

**Comment by rskops** [comment]
SonarQube analysis reported 151 issues
* ![MAJOR][MAJOR] 82 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 60 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [VM.java#](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#): This file has 1,501 lines, which is greater than 1,000 authorized. Split it into smaller files. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [VM.java#L182](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L182): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [VM.java#L209](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L209): Method org.ethereum.vm.VM.doADD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L226](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L226): Method org.ethereum.vm.VM.doMUL() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L242](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L242): Method org.ethereum.vm.VM.doSUB() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L258](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L258): Method org.ethereum.vm.VM.doDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L274](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L274): Method org.ethereum.vm.VM.doSDIV() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L290](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L290): Method org.ethereum.vm.VM.doMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L306](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L306): Method org.ethereum.vm.VM.doSMOD() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)
1. ![MAJOR][MAJOR] [VM.java#L327](https://github.com/rsksmart/rskj/blob/862b61f4bef78c0e5ad8d345c19e24bd3a237bb7/rskj-core/src/main/java/org/ethereum/vm/VM.java#L327): Method org.ethereum.vm.VM.doEXP() appears to call the same method on the same object redundantly [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**